components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java

```diff
@@ -1105,14 +1121,16 @@ public API getAPIByContextAndVersion(String context, String version, String depl
                         String revision = resultSet.getString("REVISION_UUID");
                         api.setStatus(resultSet.getString("STATUS"));
                         api.setIsDefaultVersion(isAPIDefaultVersion(connection, provider, name, version));
-                        api.setPolicy(getAPILevelTier(connection, apiUuid, revision));
-                        if (deployment.equals(deploymentName)) {
+                        if (isExpand) {
+                            api.setPolicy(getAPILevelTier(connection, apiUuid, revision));
                             if (APIConstants.API_PRODUCT.equals(apiType)) {
                                 attachURlMappingDetailsOfApiProduct(connection, api);
                                 return api;
                             }
                             attachURLMappingDetails(connection, revision, api);
                             return api;
+                        } else {
+                            api.setPolicy(null);
                         }
```

Comment:
Shall we move the return statement to here after the if block like `getApiByUUID` method?
Otherwise, if isExpand is false, this might return a "null"

Reply:
yes, done